src/main/groovy/org/shipkit/internal/gradle/plugin/PluginValidatorTask.java

```diff
@@ -0,0 +1,37 @@
+package org.shipkit.internal.gradle.plugin;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.*;
+
+import java.io.File;
+import java.util.Set;
+
+
+public class PluginValidatorTask extends DefaultTask {
+
+    @Optional
+    @Input
+    private SourceSet sourceSet;
+
+    @TaskAction
+    public void validate() {
+        Set<File> gradlePlugins;
+        Set<File> gradleProperties;
+        if (sourceSet == null) {
```

Comment:
I suggest to avoid conditional complexity and make the 'sourceSet' property mandatory instead Optional. We would set the source set in the plugin. This way, the code is simpler, easier to maintain requires us to write less tests.